app/controllers/admin_api/books_controller.rb

```diff
@@ -3,9 +3,15 @@ class AdminApi::BooksController < ApplicationController
   before_action :set_book, only: [:edit, :update, :destroy, :show]
 
   def index
-    @books = Book.search(params[:search])
-                 .order_by(params[:sort_field], params[:sort_order])
-                 .page(params[:page])
+    if params[:sort_field] == "Most popular"
+      @books = Kaminari.paginate_array(Book.search(params[:search]).most_popular)
+                       .page(params[:page])
+    else
+      @books = Book.search(params[:search])
+                   .order_by(params[:sort_field], params[:sort_order])
+                   .page(params[:page])
+                   .decorate
+    end
     @sortable_fields = Book.sort_options
   end
```

Comment:
Omg. I'm the worst. I meant to respond to this yesterday. Anyways, I just wanted to make sure I had that right. This `index` method is now doing a lot. Your code design would benefit from reducing the responsibility of this method. That could mean moving some of this into a service or simply having a private method that handles some of the work. It's definitely up to you how you want to do it but if you need any suggestions, let me know. After you've refactored this method, :shipit:.